bioblend/cloudman/launch.py

```diff
@@ -858,7 +859,12 @@ def _checkURL(self, url):
         """
         try:
             p = urlparse(url)
-            h = HTTPConnection(p[1])
+            if p.scheme == 'http':
+                h = HTTPConnection(p[1])
+            elif p.scheme == 'https':
+                h = HTTPSConnection(p[1])
+            else:
+                raise RuntimeError("Unexpected url scheme: {}".format(p.scheme))
             h.putrequest('HEAD', p[2])
             h.endheaders()
             r = h.getresponse()
```

Comment:
I am not familiar with this part of BioBlend code, but can this use the higher level `requests` that we use elsewhere?
```python
            r = requests.head(url)
            if r.status_code in (200, 401):
```

Reply:
Probably, just trying to make a minimal change least likely to affect others. Perhaps the head method was not available when the original code was written? Seems unlikely, but perhaps...

Reply:
OK, did some quick testing. The requests.head strategy hangs (due to retries?) if it gets no response from the url as would be the case as the instance is starting up - I think the original strategy is used to so that the method returns False immediately if the instance is not up yet. I'm going to tweak the patch to continue the current code behavior unless the scheme is "https" so that URLs with no scheme that used to (correctly) use HTTP continue to work. Perhaps this was the reason for the odd (to me) failures in the checks.

Reply:
From what I see with an unreachable URL under Linux, both `h.endheaders()` and `requests.head(url)` hang for a while but fail after several seconds with `[Errno 110] Connection timed out` (a `TimeoutError` and `ConnectionError` exception respectively).

Reply:
> Perhaps this was the reason for the odd (to me) failures in the checks.

If you mean the failures in https://github.com/galaxyproject/bioblend/actions/runs/802338623 , these are not connected with your changes (the failed tests were not using Cloudman and some BioBlend tests randomly fail unfortunately).

Reply:
> From what I see with an unreachable URL under Linux, both `h.endheaders()` and `requests.head(url)` hang for a while but fail after several seconds with `[Errno 110] Connection timed out` (a `TimeoutError` and `ConnectionError` exception respectively).

Hmm. Not sure why my experience was quite different. Also Linux (CentOS 7.9). With requests.head(url) it didn't return until the instance came up, whereas h.endheaders() was called and returned multiple times before the instance came up. Perhaps it is the difference between an IP address that has an instance, but doesn't yet have the cloudman app running, and an IP address which doesn't exist at all? Not sure. Regardless, in my testing the existing approach worked well, while the requests.head(url) didn't work as well.

Reply:
> 
> 
> > Perhaps this was the reason for the odd (to me) failures in the checks.
> 
> If you mean the failures in https://github.com/galaxyproject/bioblend/actions/runs/802338623 , these are not connected with your changes (the failed tests were not using Cloudman and some BioBlend tests randomly fail unfortunately).

Good to know. Thanks...